includes/SecurityUtils.php

```
@@ -51,6 +51,90 @@ public static function escapeHtml(string $content): string
+    public static function getSafeBaseUrl(): string
+    {
+        $protocol = (!empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] !== 'off') ? "https://" : "http://";
+        $host     = $_SERVER['HTTP_HOST'] ?? 'localhost';
+        $host     = preg_replace('/[^a-zA-Z0-9\-.:]/', '', $host);
+
+        // Validate host against trusted allowlist
+        $allowedHosts = ['localhost', '127.0.0.1', '::1', 'cmsfornerd.test'];
+        $hostName = explode(':', $host)[0];
+        if (!in_array($hostName, $allowedHosts, true)) {
+            http_response_code(400);
+            die("Untrusted Host header");
```

PHPMD (2.15.0)

[error] The method getSafeBaseUrl() contains an exit expression. (undefined)

(ExitExpression)